File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.py
class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        mpp={}
        cnt=0
        for i in range(len(t)):
            mpp[t[i]]=i
        for i in range(len(s)):
            if s[i] in mpp:
                cnt+=abs(i-mpp[s[i]])
        return cnt

File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        unordered_map<char,int>mpp;
        int cnt=0;
        for(int i=0;i<t.size();i++){
            mpp[t[i]]=i;
        }
        for(int i=0;i<s.size();i++){
            char ch=s[i];
            if(mpp.find(ch)!=mpp.end()){
                cnt+=abs(i-mpp[ch]);
            }
        }
        return cnt;
    }
};